src/test/java/org/junitpioneer/jupiter/NotWorkingExtensionTests.java

```diff
@@ -0,0 +1,225 @@
+				.whichAborted()
+				.withExceptionInstanceOf(TestAbortedException.class)
+				.hasMessage("Test marked as 'not working' failed as expected");
+	}
+
+	@Test
+	void abortsOnTestFailureWithMessage() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClass.class, "failureWithMessage");
+		assertThat(results)
+				.hasSingleStartedTest()
+				.whichAborted()
+				.withExceptionInstanceOf(TestAbortedException.class)
+				.hasMessage("Custom message");
+	}
+
+	@Test
+	void abortsOnError() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClass.class, "error");
+		assertThat(results)
+				.hasSingleStartedTest()
+				.whichAborted()
+				.withExceptionInstanceOf(TestAbortedException.class)
+				.hasMessage("Test marked as 'not working' failed as expected");
+	}
+
+	@Test
+	void preservesTestAbort() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClass.class, "aborted");
+		assertThat(results)
+				.hasSingleStartedTest()
+				.whichAborted()
+				.withExceptionInstanceOf(TestAbortedException.class)
+				// Ignore prefix generated by JUnit
+				.hasMessageEndingWith("custom assumption message");
+	}
+
+	@Test
+	void failsOnWorkingTest() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClass.class, "working");
+		assertThat(results)
+				.hasSingleStartedTest()
+				.whichFailed()
+				.withExceptionInstanceOf(AssertionError.class)
+				.hasMessage("Test marked as 'not working' succeeded; remove @NotWorking from it");
+	}
+
+	@Test
+	void doesNotAbortOnBeforeEachTestFailure() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClassFailureBeforeEach.class, "test");
+		assertThat(results)
+				.hasSingleFailedTest()
+				.withExceptionInstanceOf(AssertionError.class)
+				.hasMessage("before each");
+	}
+
+	@Test
+	void abortsOnAfterEachTestFailure() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClassFailureAfterEach.class, "test");
+		assertThat(results)
+				.hasSingleStartedTest()
+				.whichAborted()
+				.withExceptionInstanceOf(TestAbortedException.class)
+				.hasMessage("Test marked as 'not working' failed as expected");
+	}
+
+	@Test
+	void doesNotAbortOnBeforeAllTestFailure() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClassFailureBeforeAll.class, "test");
+		assertThat(results).hasNumberOfStartedTests(0);
+		assertThat(results)
+				.hasSingleFailedContainer()
+				.withExceptionInstanceOf(AssertionError.class)
+				.hasMessage("before all");
+	}
+
+	@Test
+	void doesNotAbortOnAfterAllTestFailure() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(DummyTestClassFailureAfterAll.class, "test");
+		assertThat(results)
+				.hasSingleStartedTest()
+				.whichFailed()
+				.withExceptionInstanceOf(AssertionError.class)
+				.hasMessage("Test marked as 'not working' succeeded; remove @NotWorking from it");
+	}
+
+	static class DummyTestClass {
```

Comment:
Can you rename this to `NotWorkingTestCases` or something else that ends with `TestCases`. 😉